app/helpers/blacklight/render_constraints_helper_behavior.rb

```diff
@@ -6,39 +6,57 @@
 # Includes methods for rendering contraints graphically on the
 # search results page (render_constraints(_*))
 module Blacklight::RenderConstraintsHelperBehavior
+  extend Deprecation
+  self.deprecation_horizon = 'blacklight 8.0'
+
   ##
   # Check if the query has any constraints defined (a query, facet, etc)
   #
   # @param [Hash] localized_params query parameters
   # @return [Boolean]
-  def query_has_constraints?(localized_params = params)
-    !(localized_params[:q].blank? && localized_params[:f].blank?)
+  def query_has_constraints?(params_or_search_state = search_state)
+    search_state = convert_to_search_state(params_or_search_state)
+    search_state.has_constraints?
   end
+  deprecation_deprecate :query_has_constraints?
 
   ##
   # Render the actual constraints, not including header or footer
   # info.
   #
   # @param [Hash] localized_params query parameters
   # @return [String]
-  def render_constraints(localized_params = params)
-    render_constraints_query(localized_params) + render_constraints_filters(localized_params)
+  def render_constraints(localized_params = params, local_search_state = search_state)
+    params_or_search_state = if localized_params != params
+                               localized_params
+                             else
+                               local_search_state
+                             end
+
+    Deprecation.silence(Blacklight::RenderConstraintsHelperBehavior) do
+      render_constraints_query(params_or_search_state) + render_constraints_filters(params_or_search_state)
+    end
   end
 
   ##
   # Render the query constraints
   #
   # @param [ActionController::Parameters] localized_params query parameters
   # @return [String]
-  def render_constraints_query(localized_params = params)
+  def render_constraints_query(params_or_search_state = search_state)
+    search_state = convert_to_search_state(params_or_search_state)
```

Comment:
Just curious, why are we doing this now?  Thanks!

Reply:
`SearchState` is nicer for applications to extend and much nicer to consume by creating a consistent interface for Blacklight to work against. It also fixes some problems we currently have where multiple plugins need to override the same methods to inject some custom parameter parsing (adv search + range limit come to mind, but it's generally true of plugins that want to have custom facet behaviors)